egui-winit/src/epi.rs

```diff
@@ -264,11 +266,17 @@ impl EpiIntegration {
     }
 
     pub fn on_event(&mut self, event: &winit::event::WindowEvent<'_>) {
-        use winit::event::WindowEvent;
-        if *event == WindowEvent::CloseRequested {
-            self.quit = self.app.on_exit_event();
-        } else if *event == WindowEvent::Destroyed {
-            self.quit = true;
+        use winit::event::{ElementState, MouseButton, WindowEvent};
+
+        match event {
+            WindowEvent::CloseRequested => self.quit = self.app.on_exit_event(),
+            WindowEvent::Destroyed => self.quit = true,
+            WindowEvent::MouseInput {
+                button: MouseButton::Left,
+                state: ElementState::Pressed,
+                ..
+            } => self.can_drag_window = true,
```

Comment:
maybe `mouse_pressed_this_frame` is a better name for this member

Reply:
I was thinking about it, but this name raises other questions:
- What button is pressed (Left, Right, or Middle)? 
- Why do we need to save this in the app context? 

Maybe something like `lmb_pressed_this_frame`or `mouse_left_pressed_this_frame` would be better? At least it will reflect the concrete button. 
But I think it's better to keep this name related to the logic. 

If you want, I can replace it with your proposal.

Reply:
This is all very nit-picky :) but:

In general, I find it's better to name things after what they signify rather than what they are used for (especially since some things are used for many things).

The line `app_output.drag_window &= self.can_drag_window;` doesn't say anything useful ("only allow drags if we allow drags"), while `app_output.drag_window &= self.lmb_pressed_this_frame;` explains exactly what is happening ("only allow drags if lmb was pressed this frame"). Anyways, it doesn't matter much, I just wanted to explain how my brain works :P

Reply:
Yeah, I got your point of view :)
As you said, it doesn't matter too much, so feel free to change it in any way. Or if you want I can't make these changes on my side.